### PR TITLE
Port demos to GNOME 46

### PR DESCRIPTION
**Pull request by sonnyp**
Closes https://github.com/workbenchdev/demos/issues/68 and more, see commits

You'll need Workbench Devel/main if you want to test this.

CI fails here because it's using Workbench 45
but I tested locally with `make ci` with Workbench 46beta https://github.com/workbenchdev/Workbench/pull/896 which runs over all demos

### CODE REVIEW

**Comment by Hofer-Julian** [comment]
>  but I tested locally with make ci in Workbench which runs over all demos

Excluding Rust demos, right?

File: demos/About Dialog/code.rs [hunk 1]
```diff
@@ -29,7 +28,7 @@ fn on_button_clicked() {
 
     dialog.add_link(
         "Documentation",
-        "https://gnome.pages.gitlab.gnome.org/libadwaita/doc/1.4/class.AboutWindow.html",
+        "https://gnome.pages.gitlab.gnome.org/libadwaita/doc/1.5/class.AboutDialog.html",
```
Comment (Hofer-Julian) [inline demos/About Dialog/code.rs:31]:
```suggestion
        "https://world.pages.gitlab.gnome.org/Rust/libadwaita-rs/stable/latest/docs/libadwaita/struct.AboutDialog.html",
```

File: demos/About Dialog/code.rs [hunk 2]
```diff
@@ -41,5 +40,5 @@ fn on_button_clicked() {
 
     dialog.add_acknowledgement_section(Some("Special thanks to"), &["My cat"]);
 
-    dialog.present();
+    dialog.present(workbench::window);
```
Comment (Hofer-Julian) [inline demos/About Dialog/code.rs:43]:
```suggestion
    dialog.present(workbench::window());
```

I am pretty sure it must be this. Did you test this example? :)

**Review by lw64 — APPROVED** [review]
Other than a few autoformatting issues (not for this pr) this looks good!

File: demos/About Dialog/code.rs (hunk 1 above)
Comment (Hofer-Julian) [reply demos/About Dialog/code.rs:31]: Sure, if you prefer that

**Review by Hofer-Julian — APPROVED** [review]

**Review by theCapypara — APPROVED** [review]

File: demos/Dialogs/main.vala [hunk 3]
```diff
@@ -1,77 +0,0 @@
-#! /usr/bin/env -S vala workbench.vala --pkg gtk4 --pkg libadwaita-1
```
Comment (theCapypara) [inline demos/Dialogs/main.vala:1]: Did you mean to just remove the Vala demo here?

File: demos/Message Dialogs/main.vala
Comment (Diego-Ivan) [inline demos/Message Dialogs/main.vala:1]: This has similar issues with the Dialogs demo (addressed in #107), but I can take care of that once this one is merged :+1:

**Review by Diego-Ivan — APPROVED** [review]

**Pull request merged**